File: backend/Backend.Contracts/DTO´s/Product/ReadProductDto.cs
using Backend.Contracts.DTO_s.Category;

namespace Backend.Contracts.DTO_s.Product;

public class ReadProductDto
{
    public Guid Id { get; set; }

    public string Name { get; set; } = string.Empty;

    public string Description { get; set; } = string.Empty;

    public decimal Price { get; set; }

    public ReadCategoryDto? Category { get; set; }

    public string? ImageUrl { get; set; }

    public int StockQuantity { get; set; }

    public string? SKU { get; set; }

    public bool IsActive { get; set; }

    public DateTime CreatedAt { get; set; }

    public DateTime? UpdatedAt { get; set; }
}

File: frontend/src/app/shared/DTO´s/Product/ReadProductDto.ts
import { AutoMap } from '@automapper/classes';
import { ReadCategoryDto } from '../Category/ReadCategoryDto';

export class ReadProductDto {
  @AutoMap()
  id: string = '';
  @AutoMap()
  name: string = '';
  @AutoMap()
  description: string = '';
  @AutoMap()
  price: number = 0;
  @AutoMap(() => ReadCategoryDto)
  category: ReadCategoryDto = new ReadCategoryDto();
  @AutoMap()
  imageUrl?: string = '';
  @AutoMap()
  stockQuantity: number = 0;
  @AutoMap()
  sku?: string = '';
  @AutoMap()
  isActive: boolean = true;
  @AutoMap()
  createdAt: Date = new Date();
  @AutoMap()
  updatedAt: Date = new Date();
}
